target/linux/ramips/mt7621/base-files/lib/upgrade/xiaomi.sh

```diff
@@ -0,0 +1,459 @@
+# SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+# shellcheck disable=SC1091
+# shellcheck disable=SC2155
+
+. /lib/functions.sh
+
+MAGIC_XIAOMI_HDR1="48445231"     # "HDR1" - xiaomi image header
+MAGIC_XIAOMI_BLK="beba0000"
+MAGIC_UIMAGE="27051956"          # uImage header
+MAGIC_UBI="55424923"             # "UBI#"
+#MAGIC_UBIFS="31181006"
+MAGIC_HSQS="68737173"            # "hsqs"
+MAGIC_SYSUPG="7379737570677261"  # TAR "sysupgrade"
+
+XIAOMI_PAGESIZE=2048
+
+XIAOMI_FW_FILE=""
+XIAOMI_FW_SIZE=0
+XIAOMI_KERNEL_PART=$CI_KERNPART
+XIAOMI_KERNEL2_PART=""
+XIAOMI_KERNEL2_NAMES="kernel_stock|kernel_dup"
+XIAOMI_ROOTFS_PART=$CI_UBIPART
+XIAOMI_ROOTFS_PARTSIZE=
+
+XIAOMI_RESTORE_ROOTFS2=
+
+log_msg() {
+	echo "$@"
+}
+
+log_err() {
+	echo "ERROR: $*" >&2
+}
+
+die() {
+	log_err "$@"
+	exit 1
+}
+
+get_uint32_at() {
+	local offset=$1
+	local endianness=$2
+	local hex
+	if [ $(( offset + 4 )) -gt $XIAOMI_FW_SIZE ]; then
+		echo ""
+		return
+	fi
+	local dd_args="if=$XIAOMI_FW_FILE skip=$offset bs=1 count=4"
+	if [ "$endianness" = "be" ]; then
+		# shellcheck disable=SC2086
+		hex=$( dd $dd_args 2>/dev/null | hexdump -v -n 4 -e '1/1 "%02x"' )
+	else
+		# shellcheck disable=SC2086
+		hex=$( dd $dd_args 2>/dev/null | hexdump -v -e '1/4 "%02x"' )
```

Comment:
```suggestion
		local dd="$(dd if="$XIAOMI_FW_FILE" skip="$offset" bs=1 count=4 2>/dev/null)"
	if [ "$endianness" = "be" ]; then
		hex=$( echo "$dd" | hexdump -v -n 4 -e '1/1 "%02x"' )
	else
		hex=$( echo "$dd" | hexdump -v -e '1/4 "%02x"' )
	fi
```

Can you check if this also work on low endian ?